File: src/domain/cart.domain.ts
export class CartDomain {
  id: string | null;
  createdAt: Date | null;
  updatedAt: Date | null;

  constructor(
    public userId: string,
    public name: string,
    public pickNum: number,
    public pickItemIds: string[],
    public pickImages: string[]
  ) {
    this.id = null;
    this.createdAt = null;
    this.updatedAt = null;
  }
}

File: src/main/java/kr/okku/server/domain/CartDomain.java
package kr.okku.server.domain;

import lombok.Builder;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
@Builder
public class CartDomain {
    private String id;
    private String userId;
    private String name;
    private int pickNum;

    private int orderIndex;
    private List<String> pickItemIds;

    public void deletePicks(List<String> pickIds){
        List<String> modifiablePickItemIds = new ArrayList<>(this.pickItemIds);
        modifiablePickItemIds.removeAll(pickIds);
        this.pickItemIds = modifiablePickItemIds;
        this.pickNum = modifiablePickItemIds.size();
    }

    public void addPicks(List<String> pickIds){
        List<String> modifiablePickItemIds = new ArrayList<>(this.pickItemIds);
        List<String> pickIdForAdd = pickIds.stream()
                .filter(pickId -> !modifiablePickItemIds.contains(pickId))
                .toList();
        modifiablePickItemIds.addAll(pickIdForAdd);
        this.pickItemIds = modifiablePickItemIds;
        this.pickNum = modifiablePickItemIds.size();
    }
}